### PR TITLE
LPS-182465 | Filter sort and search the list of applications

### PR DESCRIPTION
**Pull request by Hijky**
**Background:**
Add a testcase to filter sort and search the list of applications

**Test Plan**
CanFilterAPIsByStatus
When I filter by Published status
Then only the published API is listed

CanFilterAPIsByExcludeStatus
When I filter by Published status with exclude enabled
Then two unpublished APIs are listed

CanFilterAPIsByLastUpdated
When I filter by Last updated From 06/30/2023 to 06/30/2023
Then I can see no API Application Found

CanSortAPIsByTitle
When I sort the APIs by Title
Then the APIs are sorted by Title

CanSortAPIsByLastUpdated
When I sort the APIs by Last Updated
Then the APIs are sorted by Last Updated

CanSortAPIsByStatus
When I sort the APIs by Status
Then the APIs are sorted by Status

CanSearchAPIsWithoutAvailableOnes
When I search the API applications with unrelated word
Then no API application were found

CanSearchAPIsByTitle
When I search the API applications with a valid title
Then I can retrieve all available API applications

**Jira ticket:**
https://liferay.atlassian.net/browse/LPS-189822

### CODE REVIEW

**Comment by Hijky** [comment]
ci:test:sf

**Comment by Hijky** [comment]
ci:test:headless-functional:nocompile

**Comment by Hijky** [comment]
The failures
LocalFile.FilterSortAndSearchTheListOfAPIApplications#CanSortAPIsByStatus
LocalFile.CreateAPIApplication#CanCreateAPIOnVirtualInstance
LocalFile.GeneratePathStructureForEndpoints#CanLoadAPIApplicationURL
related to this [ticket](https://liferay.atlassian.net/browse/LPS-193414)

LocalFile.CreateAPIApplication#CanCreateApiWithValidFieldValues will be fixed in this [PR](https://github.com/liferay-headless/liferay-portal/pull/1452)

the failure Element is present at "//html[contains(@class,'lfr-spa-loading')]" is analysising

**Review by magjed4289 — APPROVED** [review]

File: modules/apps/headless/headless-builder/headless-builder-test/src/testFunctional/tests/[UI]AllowUsersToDefineAPIApplicationsInTheAPIBuilder/FilterSortAndSearchTheListOfAPIApplications.testcase [hunk 1]
```diff
@@ -0,0 +1,230 @@
...
+	}
+
+	@priority = 4
+	test CanFilterAPIsByExcludeStatus {
+		property portal.acceptance = "true";
+		property test.liferay.virtual.instance = "false";
+		property test.run.type = "single";
+
+		task ("When I filter by Published status with exclude enabled") {
+			APIBuilderUI.filterByStatus(
```
[the remark this answers is not in the document]
Comment (Hijky) [reply modules/apps/headless/headless-builder/headless-builder-test/src/testFunctional/tests/[UI]AllowUsersToDefineAPIApplicationsInTheAPIBuilder/FilterSortAndSearchTheListOfAPIApplications.testcase:55]: the exclude default status is disabled

File: modules/apps/headless/headless-builder/headless-builder-test/src/testFunctional/tests/[UI]AllowUsersToDefineAPIApplicationsInTheAPIBuilder/FilterSortAndSearchTheListOfAPIApplications.testcase [hunk 2]
```diff
@@ -0,0 +1,230 @@
...
+	test CanSearchAPIsWithoutAvailableOnes {
+		property portal.acceptance = "true";
+		property test.liferay.virtual.instance = "false";
+		property test.run.type = "single";
+
+		task ("When I search the API applications with unrelated word") {
+			APIBuilderUI.searchItem(value = "Application");
+		}
+
+		task ("Then no API application were found") {
```
Comment (yanan) [inline modules/apps/headless/headless-builder/headless-builder-test/src/testFunctional/tests/[UI]AllowUsersToDefineAPIApplicationsInTheAPIBuilder/FilterSortAndSearchTheListOfAPIApplications.testcase:161]: Same here, I think we can assert "No API Application Found" here.

Comment (yanan) [reply modules/apps/headless/headless-builder/headless-builder-test/src/testFunctional/tests/[UI]AllowUsersToDefineAPIApplicationsInTheAPIBuilder/FilterSortAndSearchTheListOfAPIApplications.testcase:161]: The codes here are correct, I just think we can use less code instead :smile:

**Review by yanan — COMMENTED** [review]
Others all good

File: modules/apps/headless/headless-builder/headless-builder-test/src/testFunctional/tests/[UI]AllowUsersToDefineAPIApplicationsInTheAPIBuilder/FilterSortAndSearchTheListOfAPIApplications.testcase [hunk 3]
```diff
@@ -0,0 +1,230 @@
...
+		property test.liferay.virtual.instance = "false";
+		property test.run.type = "single";
+
+		task ("When I filter by Last updated From 06/30/2023 to 06/30/2023") {
+			APIBuilderUI.filterByLastUpdated(
+				fromDate = "06/30/2023",
+				toDate = "06/30/2023");
+		}
+
+		task ("Then I can see no API Application Found") {
```
[the remark this answers is not in the document]
Comment (Hijky) [reply modules/apps/headless/headless-builder/headless-builder-test/src/testFunctional/tests/[UI]AllowUsersToDefineAPIApplicationsInTheAPIBuilder/FilterSortAndSearchTheListOfAPIApplications.testcase:90]: you are right, will change it

**Comment by Hijky** [comment]
ci:test:sf

**Comment by Hijky** [comment]
ci:test:headless-functional:nocompile

**Comment by Hijky** [comment]
The failures are not related to PR, and I will analyze the failures

**Comment by Hijky** [comment]
ci:forward

**Pull request closed without merging**